cpmpy/solvers/gurobi.py

```diff
@@ -181,10 +181,10 @@ def solve(self, time_limit=None, solution_callback=None, **kwargs):
             # set _objective_value
             if self.has_objective():
                 grb_obj_val = grb_objective.getValue()
-                if grb_obj_val != int(grb_obj_val):
-                    self.objective_value_ =  grb_obj_val # can happen with DirectVar using floats
-                else:
+                if round(grb_obj_val) == grb_obj_val: # it is an integer?:
                     self.objective_value_ = int(grb_obj_val)
```

Comment:
do we need this? this example shows the inconsistency between our value function and objective value calculation:
m = Model()
m += 2 * x + 3 * y == 2
m.maximize(5*x + 0.7*y)
m.solve()
s = SolverLookup.get('ortools', m)
print(m.objective_value())
s.solve()
print(s.objective_value())

This happens because m.objective_value() just does m.objective_.value(), it does not know about the converted objective value in s.

Reply:
This should be fixed now with the extra check.

Reply:
If we do not return an integer here, you sometimes get quite confusing errors afterwards.
For example, if I want to make a model UNSAT by restriciting the objective value

```python
s = cp.SolverLookup.get("gurobi")
obj = ...
s.minimize(obj)
assert s.solve()
s += obj.value() < s.objective_value()
assert s.solve() is False # will throw an error if objective_value() returns a float
```